skopt/space.py

```diff
@@ -0,0 +1,458 @@
+            Set random state to something other than None for reproducible
+            results.
+
+        Returns
+        -------
+        * `points`: [array-like, shape=(n_points, n_dims)]
+           Points sampled from the space.
+        """
+        rng = check_random_state(random_state)
+
+        # Draw
+        columns = []
+
+        for dim in self.dimensions:
+            if sp_version < (0, 16):
+                columns.append(dim.rvs(n_samples=n_samples))
+            else:
+                columns.append(dim.rvs(n_samples=n_samples, random_state=rng))
+
+        # Transpose
+        rows = []
+
+        for i in range(n_samples):
+            r = []
+            for j in range(self.n_dims):
+                r.append(columns[j][i])
+
+            rows.append(r)
+
+        return np.asarray(rows)
+
+    def transform(self, X):
+        """Transform samples from the original space into a warped space.
+
+        Note: this transformation is expected to be used to project samples
+              in a space where numerical optimization is suited.
+
+        Parameters
+        ----------
+        * `X` [array-like, shape=(n_samples, n_dims)]:
+            The samples to transform.
+
+        Returns
+        -------
+        * `Xt` [array-like, shape=(n_samples, transformed_n_dims)]
+            The transformed samples.
+        """
+        # Pack by dimension
+        columns = []
+        for dim in self.dimensions:
+            columns.append([])
+
+        for i in range(len(X)):
+            for j in range(self.n_dims):
+                columns[j].append(X[i][j])
+
+        # Transform
+        for j in range(self.n_dims):
+            columns[j] = self.dimensions[j].transform(columns[j])
+
+        # Repack as an array
+        Xt = np.hstack([np.asarray(c).reshape((len(X), -1)) for c in columns])
+
+        return Xt
+
+    def inverse_transform(self, Xt):
+        """Inverse transform samples from the warped space back to the
+           original space.
+
+        Parameters
+        ----------
+        * `Xt` [array-like, shape=(n_samples, transformed_n_dims)]:
+            The samples to inverse transform.
+
+        Returns
+        -------
+        * `X` [array-like, shape=(n_samples, n_dims)]
+            The original samples.
+        """
+        # Inverse transform
+        columns = []
+        start = 0
+
+        for j in range(self.n_dims):
+            dim = self.dimensions[j]
+            offset = dim.transformed_size
+
+            if offset == 1:
+                columns.append(dim.inverse_transform(Xt[:, start]))
+            else:
+                columns.append(
+                    dim.inverse_transform(Xt[:, start:start+offset]))
+
+            start += offset
+
+        # Transpose
+        rows = []
+
+        for i in range(len(Xt)):
+            r = []
+            for j in range(self.n_dims):
+                r.append(columns[j][i])
+
+            rows.append(r)
+
+        return np.asarray(rows)
```

Comment:
`np.transpose(columns)` does not work? (Neglecting the fact that it will return an object dtype)